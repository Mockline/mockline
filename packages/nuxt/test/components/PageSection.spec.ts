import { describe, it, expect } from 'vitest';
import ComponentRender from "../component-render"
import type { TypeOf } from 'zod'
// @ts-ignore
import { MPageSection } from '#components'

describe('PageSection', () => {
  it.each([
    ['base case', {}],
    ['with title and description', {
      title: 'Beautiful Vue UI components',
      description: 'Mockline provides a comprehensive suite of components and utilities to help you build beautiful and accessible web applications with Vue and Nuxt.'
    }],
    ['with headline and icon', {
      headline: 'Features',
      icon: 'lucide:rocket',
      title: 'Beautiful Vue UI components'
    }],
    ['with headline, title and description', {
      headline: 'Features',
      title: 'Beautiful Vue UI components',
      description: 'Mockline provides a comprehensive suite of components and utilities to help you build beautiful and accessible web applications with Vue and Nuxt.'
    }],
    ['with left alignment', {
      headline: 'Features',
      title: 'Beautiful Vue UI components',
      description: 'Mockline provides a comprehensive suite of components and utilities to help you build beautiful and accessible web applications with Vue and Nuxt.',
      align: 'left'
    }],
    ['with horizontal orientation and reverse', {
      headline: 'Features',
      description: 'Mockline provides a comprehensive suite of components and utilities to help you build beautiful and accessible web applications with Vue and Nuxt.',
      orientation: 'horizontal',
      reverse: true
    }]
  ])('renders %s correctly', async (nameOrHtml: string, options?: TypeOf<typeof MPageSection.props>) => {
    const html = await ComponentRender(nameOrHtml, options, MPageSection)
    expect(html).toMatchSnapshot()
  })

  // Test avec des slots
  it('renders with custom title slot', async () => {
    const html = await ComponentRender('with custom title slot', {
      headline: 'Features',
      description: 'Mockline provides a comprehensive suite of components and utilities to help you build beautiful and accessible web applications with Vue and Nuxt.',
      align: 'center',
      orientation: 'horizontal',
      reverse: true
    }, MPageSection, {
      title: '<h2 class="text-3xl font-bold text-blue-200">Beautiful Vue UI components</h2>'
    })
    expect(html).toMatchSnapshot()
  })

  // Test avec image dans le slot default
  it('renders with image in default slot', async () => {
    const html = await ComponentRender('with image in default slot', {
      headline: 'Features',
      description: 'Mockline provides a comprehensive suite of components and utilities to help you build beautiful and accessible web applications with Vue and Nuxt.',
      align: 'center',
      orientation: 'horizontal',
      reverse: true
    }, MPageSection, {
      title: '<h2 class="text-3xl font-bold text-blue-200">Beautiful Vue UI components</h2>',
      default: '<img src="https://picsum.photos/704/1294" alt="Illustration" class="rounded-lg max-w-[400px] object-cover">'
    })
    expect(html).toMatchSnapshot()
  })

  // Test pour l'orientation horizontale avec image
  it('renders horizontal orientation with image correctly', async () => {
    const html = await ComponentRender('horizontal orientation with image', {
      headline: 'Features',
      title: 'Beautiful Vue UI components',
      description: 'Mockline provides a comprehensive suite of components and utilities to help you build beautiful and accessible web applications with Vue and Nuxt.',
      orientation: 'horizontal'
    }, MPageSection, {
      default: '<img src="https://picsum.photos/704/1294" alt="Illustration" class="rounded-lg max-w-[400px] object-cover">'
    })
    expect(html).toMatchSnapshot()
  })

  // Test pour l'orientation verticale avec image
  it('renders vertical orientation with image correctly', async () => {
    const html = await ComponentRender('vertical orientation with image', {
      headline: 'Features',
      title: 'Beautiful Vue UI components',
      description: 'Mockline provides a comprehensive suite of components and utilities to help you build beautiful and accessible web applications with Vue and Nuxt.',
      orientation: 'vertical'
    }, MPageSection, {
      default: '<img src="https://picsum.photos/704/1294" alt="Illustration" class="rounded-lg max-w-[400px] object-cover">'
    })
    expect(html).toMatchSnapshot()
  })
})
