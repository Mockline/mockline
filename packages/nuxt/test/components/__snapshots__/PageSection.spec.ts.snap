// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PageSection > renders base case correctly 1`] = `
"<div class="relative isolate">
  <div class="flex py-16 sm:py-24 lg:py-32 gap-8 sm:gap-16 lg:grid lg:grid-cols-2 lg:items-center flex-row">
    <!--v-if-->
    <!--v-if-->
  </div>
</div>"
`;

exports[`PageSection > renders center alignment correctly 1`] = `
"<div class="relative isolate">
  <div class="flex py-16 sm:py-24 lg:py-32 gap-8 sm:gap-16 lg:grid lg:grid-cols-2 lg:items-center flex-row">
    <!--v-if-->
    <!--v-if-->
  </div>
</div>"
`;

exports[`PageSection > renders complete example correctly 1`] = `
"<div class="relative isolate">
  <div class="flex py-16 sm:py-24 lg:py-32 gap-8 sm:gap-16 lg:grid lg:grid-cols-2 lg:items-center flex-row">
    <!--v-if-->
    <!--v-if-->
  </div>
</div>"
`;

exports[`PageSection > renders horizontal images correctly 1`] = `
"<div class="relative isolate">
  <div class="flex py-16 sm:py-24 lg:py-32 gap-8 sm:gap-16 lg:grid lg:grid-cols-2 lg:items-center flex-row">
    <!--v-if-->
    <!--v-if-->
  </div>
</div>"
`;

exports[`PageSection > renders horizontal orientation correctly 1`] = `
"<div class="relative isolate">
  <div class="flex py-16 sm:py-24 lg:py-32 gap-8 sm:gap-16 lg:grid lg:grid-cols-2 lg:items-center flex-row">
    <!--v-if-->
    <!--v-if-->
  </div>
</div>"
`;

exports[`PageSection > renders horizontal orientation with image correctly 1`] = `
"<div class="relative isolate">
  <div class="flex py-16 sm:py-24 lg:py-32 gap-8 sm:gap-16 lg:grid lg:grid-cols-2 lg:items-center flex-row">
    <!--v-if-->
    <!--v-if-->
  </div>
</div>"
`;

exports[`PageSection > renders images with correct orientation 1`] = `
"<div class="relative isolate">
  <div class="flex py-16 sm:py-24 lg:py-32 gap-8 sm:gap-16 lg:grid lg:grid-cols-2 lg:items-center flex-row">
    <!--v-if-->
    <!--v-if-->
  </div>
</div>"
`;

exports[`PageSection > renders images with correct orientation 2`] = `
"<div class="relative isolate">
  <div class="flex py-16 sm:py-24 lg:py-32 gap-8 sm:gap-16 lg:grid lg:grid-cols-2 lg:items-center flex-row">
    <!--v-if-->
    <!--v-if-->
  </div>
</div>"
`;

exports[`PageSection > renders left alignment correctly 1`] = `
"<div class="relative isolate">
  <div class="flex py-16 sm:py-24 lg:py-32 gap-8 sm:gap-16 lg:grid lg:grid-cols-2 lg:items-center flex-row">
    <!--v-if-->
    <!--v-if-->
  </div>
</div>"
`;

exports[`PageSection > renders right alignment correctly 1`] = `
"<div class="relative isolate">
  <div class="flex py-16 sm:py-24 lg:py-32 gap-8 sm:gap-16 lg:grid lg:grid-cols-2 lg:items-center flex-row">
    <!--v-if-->
    <!--v-if-->
  </div>
</div>"
`;

exports[`PageSection > renders vertical images correctly 1`] = `
"<div class="relative isolate">
  <div class="flex py-16 sm:py-24 lg:py-32 gap-8 sm:gap-16 lg:grid lg:grid-cols-2 lg:items-center flex-row">
    <!--v-if-->
    <!--v-if-->
  </div>
</div>"
`;

exports[`PageSection > renders vertical orientation correctly 1`] = `
"<div class="relative isolate">
  <div class="flex py-16 sm:py-24 lg:py-32 gap-8 sm:gap-16 lg:grid lg:grid-cols-2 lg:items-center flex-row">
    <!--v-if-->
    <!--v-if-->
  </div>
</div>"
`;

exports[`PageSection > renders vertical orientation with image correctly 1`] = `
"<div class="relative isolate">
  <div class="flex py-16 sm:py-24 lg:py-32 gap-8 sm:gap-16 lg:grid lg:grid-cols-2 lg:items-center flex-row">
    <!--v-if-->
    <!--v-if-->
  </div>
</div>"
`;

exports[`PageSection > renders with custom slots 1`] = `
"<div class="relative isolate">
  <div class="flex py-16 sm:py-24 lg:py-32 gap-8 sm:gap-16 lg:grid lg:grid-cols-2 lg:items-center flex-row">
    <!--v-if-->
    <!--v-if-->
  </div>
</div>"
`;

exports[`PageSection > renders with custom title slot 1`] = `
"<div class="relative isolate">
  <div class="flex py-16 sm:py-24 lg:py-32 gap-8 sm:gap-16 lg:grid lg:grid-cols-2 lg:items-center flex-row">
    <!--v-if-->
    <!--v-if-->
  </div>
</div>"
`;

exports[`PageSection > renders with headline and icon correctly 1`] = `
"<div class="relative isolate">
  <div class="flex py-16 sm:py-24 lg:py-32 gap-8 sm:gap-16 lg:grid lg:grid-cols-2 lg:items-center flex-row">
    <!--v-if-->
    <!--v-if-->
  </div>
</div>"
`;

exports[`PageSection > renders with headline, title and description correctly 1`] = `
"<div class="relative isolate">
  <div class="flex py-16 sm:py-24 lg:py-32 gap-8 sm:gap-16 lg:grid lg:grid-cols-2 lg:items-center flex-row">
    <!--v-if-->
    <!--v-if-->
  </div>
</div>"
`;

exports[`PageSection > renders with horizontal orientation and reverse correctly 1`] = `
"<div class="relative isolate">
  <div class="flex py-16 sm:py-24 lg:py-32 gap-8 sm:gap-16 lg:grid lg:grid-cols-2 lg:items-center flex-row">
    <!--v-if-->
    <!--v-if-->
  </div>
</div>"
`;

exports[`PageSection > renders with image in default slot 1`] = `
"<div class="relative isolate">
  <div class="flex py-16 sm:py-24 lg:py-32 gap-8 sm:gap-16 lg:grid lg:grid-cols-2 lg:items-center flex-row">
    <!--v-if-->
    <!--v-if-->
  </div>
</div>"
`;

exports[`PageSection > renders with left alignment correctly 1`] = `
"<div class="relative isolate">
  <div class="flex py-16 sm:py-24 lg:py-32 gap-8 sm:gap-16 lg:grid lg:grid-cols-2 lg:items-center flex-row">
    <!--v-if-->
    <!--v-if-->
  </div>
</div>"
`;

exports[`PageSection > renders with props and slots combined 1`] = `
"<div class="relative isolate">
  <div class="flex py-16 sm:py-24 lg:py-32 gap-8 sm:gap-16 lg:grid lg:grid-cols-2 lg:items-center flex-row">
    <!--v-if-->
    <!--v-if-->
  </div>
</div>"
`;

exports[`PageSection > renders with reverse order correctly 1`] = `
"<div class="relative isolate">
  <div class="flex py-16 sm:py-24 lg:py-32 gap-8 sm:gap-16 lg:grid lg:grid-cols-2 lg:items-center flex-row">
    <!--v-if-->
    <!--v-if-->
  </div>
</div>"
`;

exports[`PageSection > renders with title and description correctly 1`] = `
"<div class="relative isolate">
  <div class="flex py-16 sm:py-24 lg:py-32 gap-8 sm:gap-16 lg:grid lg:grid-cols-2 lg:items-center flex-row">
    <!--v-if-->
    <!--v-if-->
  </div>
</div>"
`;
